### PR TITLE
LOG-3302: Update scc to desired

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* fixes updating by Updating the current SCC to desired

### Links
* https://issues.redhat.com/browse/LOG-3302

### CODE REVIEW

File: internal/reconcile/scc.go [hunk 1]
```diff
@@ -27,7 +27,37 @@ func SecurityContextConstraints(k8Client client.Client, desired *security.Securi
...
+	to.RunAsUser = from.RunAsUser
+	to.SupplementalGroups = from.SupplementalGroups
+	to.FSGroup = from.FSGroup
+	to.ReadOnlyRootFilesystem = from.ReadOnlyRootFilesystem
+	to.Users = from.Users
+	to.Groups = from.Groups
+	to.SeccompProfiles = from.SeccompProfiles
+	to.AllowedUnsafeSysctls = from.AllowedUnsafeSysctls
+	to.ForbiddenSysctls = from.ForbiddenSysctls
+
```
Comment (vimalk78) [inline internal/reconcile/scc.go:61]: can we replace this block with `*to=*from` if all fields are to be copied?

Comment (jcantrill) [reply internal/reconcile/scc.go:61]: SCCs don't have a "spec" field, they live off the root of the type. The only change we possibly could make is to use reflection over the fields

Comment (vimalk78) [reply internal/reconcile/scc.go:61]: agree. a `*to=*from` would copy more fields than we need.

**Pull request merged**